lib/tracecontext/index.js

```diff
@@ -63,8 +63,17 @@ class TraceContext {
   toTraceParentString () {
     return this.traceparent.toString()
   }
+
+  propagateTraceContextHeaders (carrier, contextFrom, setter) {
+    return propagateTraceContextHeaders(carrier, contextFrom, setter)
+  }
+}
+
+function propagateTraceContextHeaders (carrier, contextFrom, setter) {
+  setter(carrier, 'traceparent', contextFrom.toTraceParentString())
+  setter(carrier, 'tracestate', contextFrom.toTraceStateString())
 }
```

Comment:
This encapsulates the knowledge of the header names to this class and only calls setters if the values are non-empty so that the setters don't have to check for that.

I'm not sure how TraceState can access `agent._conf.useElasticTraceparentHeader`. Probably best to pass in `agent._conf.useElasticTraceparentHeader` as a constructor argument.

```suggestion
  propagateTraceContextHeaders (carrier, contextFrom, setter) {
    const traceparent = contextFrom.toTraceParentString()
    if (traceparent) {
      setter(carrier, 'traceparent', traceparent)
    }
    if (agent._conf.useElasticTraceparentHeader) {
      setter(carrier, 'elastic-apm-traceparent', traceparent)
    }
    const tracestate = contextFrom.toTraceStateString()
    if (tracestate) {
      setter(carrier, 'tracestate', contextFrom.toTraceStateString())
    }
  }
}
```